### PR TITLE
[fixes #132] Don't treat headers as regular rows (so style can be app…

### LINKED ISSUE #132: Why is header not added as header in XLS ?
Why are headers not added as headers (via `->addHeaders(...)`) at https://github.com/alperenersoy/filament-export/blob/main/src/FilamentExport.php#L136 and why is `->noHeaderRow()` enabled ? Is it because of compatibility with CSV ? Would be here some room for distinguishing them so we can set eg. headerStyle via `modifyExcelWriter` ?

### PR DESCRIPTION
**Pull request by lukasinko**
…lied to the header) in case of XLS

### CODE REVIEW

**Comment by lukasinko** [comment]
@alperenersoy Could you give a look ? Thank you

**Comment by lukasinko** [comment]
One can then do things like 
```
                    ->modifyExcelWriter(function(SimpleExcelWriter $excelWriter){
                        $options = $excelWriter->getWriter()->getOptions();
                        $options->DEFAULT_COLUMN_WIDTH=25; // set default width
                        $options->DEFAULT_ROW_HEIGHT=15; // set default height
                        // set columns 1, 3 and 8 to width 40
                        $options->setColumnWidth(40, 1, 3, 8);
                        // set columns 9 through 12 to width 10
                        $options->setColumnWidthForRange(10, 9, 12);


                        $border = new Border(
                            new BorderPart(Border::BOTTOM, Color::LIGHT_BLUE, Border::WIDTH_THIN, Border::STYLE_SOLID),
                            new BorderPart(Border::LEFT, Color::LIGHT_BLUE, Border::WIDTH_THIN, Border::STYLE_SOLID),
                            new BorderPart(Border::RIGHT, Color::LIGHT_BLUE, Border::WIDTH_THIN, Border::STYLE_SOLID),
                            new BorderPart(Border::TOP, Color::LIGHT_BLUE, Border::WIDTH_THIN, Border::STYLE_SOLID)
                        );

                        $style = (new Style())
                            ->setFontBold()
                            ->setFontSize(15)
                            ->setFontColor(Color::BLUE)
                            ->setShouldWrapText()
                            ->setBackgroundColor(Color::YELLOW)
                            ->setBorder($border);

                        $excelWriter->setHeaderStyle($style);
                        return $excelWriter;
                    } )
```.

Adding specific modification functions like modifyStyleUsing, modifyOptionsUsing is maybe too verbose to me, if we alrady have modifyExcelWriter. I would have such functions if there would be high demand for them.

**Comment by lukasinko** [comment]
@alperenersoy Would you mind to merge it ? Thank you in advance.

**Comment by alperenersoy** [comment]
@lukasinko I asked sth in review but I think you didn't see that. Can you check the question, please?

**Comment by lukasinko** [comment]
@alperenersoy I don't see any review :-O Have you submitted it ?

File: src/FilamentExport.php [hunk 1]
```diff
@@ -132,14 +132,20 @@ public function download(): StreamedResponse
...
-            $stream = SimpleExcelWriter::streamDownload("{$this->getFileName()}.{$this->getFormat()}", $this->getFormat(), delimiter: $this->getCsvDelimiter())
-                ->noHeaderRow()
-                ->addRows($this->getRows()->prepend($headers));
+            $stream = SimpleExcelWriter::streamDownload("{$this->getFileName()}.{$this->getFormat()}", $this->getFormat(), delimiter: $this->getCsvDelimiter());
 
             if ($modifyExcel = $this->getModifyExcelWriter()) {
                 $stream = $modifyExcel($stream);
             }
 
+            if ($this->getFormat() === "csv") {
```
[the remark this answers is not in the document]
Comment (lukasinko) [reply src/FilamentExport.php:141]: I can try it, i didn't want to touch area i don't understand much (eg. CSV in combination with noHeaderRow + prepend).

[the remark this answers is not in the document]
Comment (alperenersoy) [reply src/FilamentExport.php:141]: I tried it before the review and it worked. You can touch it.

[the remark this answers is not in the document]
Comment (lukasinko) [reply src/FilamentExport.php]: Done. Thank you.

**Comment by alperenersoy** [comment]
@lukasinko merged this pr and the other one to both versions and released them. thanks for your contribution. appreciate it.

**Pull request merged**